diff -r ff7b35837d0f runner.py
--- a/runner.py	Sat Jul 21 13:35:54 2012 +0200
+++ b/runner.py	Mon Jul 23 16:22:08 2012 +0200
@@ -28,7 +28,7 @@
     funcs = perf.BENCH_FUNCS.copy()
     funcs.update(perf._FindAllBenchmarks(benchmarks.__dict__))
     opts = ['-b', ','.join(benchmark_set),
-            '--inherit_env=PATH',
+            '--inherit_env=PATH,PYPYLOG',
             '--no_charts']
     if fast:
         opts += ['--fast']
